### PR TITLE
GLTFExporter: Support individual morph target animation.

### PR DESCRIPTION
**Pull request by donmccurdy**
- ~~Adds AnimationUtils.insertKeyframe().~~
- ~~Adds AnimationUtils.mergeMorphTargetTracks().~~
- Adds AnimationClip.prototype.clone().
- Adds KeyframeTrack.prototype.clone().

While three.js allows animation clips to manipulate a single morph target at a time, glTF currently does not. This PR enables GLTFExporter to automatically merge animation tracks affecting individual morph targets, and write filler data if needed. I used a version of this code to complete #14747, converting the Rome models to GLB, which relied heavily on morph target animation.

The addition of `track.clone()` and `clip.clone()` provide a methods I've needed before — both also appear in https://github.com/mrdoob/three.js/pull/13430.

/cc @takahirox @fernandojsg

### CODE REVIEW

**Comment by takahirox** [comment]
I locally do the similar thing to convert from MMD to glTF. This's very helpful for sure.

Can't we move `mergeMorphTargetTracks` to `AnimationUtils`?

**Comment by donmccurdy** [comment]
Yeah we can certainly move `mergeMorphTargetTracks` to `AnimationUtils`. That does mean it would be in `src/` rather than `examples/js`, I just wasn't sure if that was OK.

**Comment by takahirox** [comment]
I'm ok with that but it's very up to @mrdoob. IMO at least `mergeMorphTargetTracks` would be better out of the exporter for the reusability of the function and the simplicity of the exporter.

File: examples/js/exporters/GLTFExporter.js [hunk 1]
```diff
@@ -277,6 +277,105 @@ THREE.GLTFExporter.prototype = {
...
+				var mergedKeyframeIndex = 0;
+
+				mergedTrack = mergedTracks[ sourceTrackNode.uuid ];
+
+				for ( var j = 0; j < sourceTrack.times.length; j ++ ) {
+
+					var keyframeIndex = THREE.AnimationUtils.insertKeyframe( mergedTrack, sourceTrack.times[ j ] );
+					mergedTrack.values[ keyframeIndex * targetCount + targetIndex ] = sourceTrack.values[ j ];
+
+				}
```
Comment (donmccurdy) [inline examples/js/exporters/GLTFExporter.js]: Hm just realized this isn't quite right, even though it was sufficient for the Rome models. If `mergedTrack` and `sourceTrack` contain:

```
mergedTrack:
0 0 0 0 // frame 1 
0 0 0 0 // frame 2
0 0 0 0 // ...
0 0 0 0
0 0 0 0

sourceTrack:
0 // frame 1
1 // frame 4
0 // frame 5
```

... we need to overwrite frames 2-3 with interpolated values from `sourceTrack`, not just overwrite frame 0, 4, and 5. Will fix this...

**Comment by donmccurdy** [comment]
I've moved `mergeMorphTargetTracks()` into AnimationUtils, and fixed the issue in my earlier comment. If we'd rather keep the extra code in GLTFExporter I'm also fine with that.

File: examples/js/exporters/GLTFExporter.js (hunk 1 above)
Comment (donmccurdy) [reply examples/js/exporters/GLTFExporter.js]: This is fixed. ✅

**Comment by donmccurdy** [comment]
@takahirox other than the remaining question of whether `insertKeyframe` and `mergeMorphTargetTracks` should be in GLTFExporter or AnimationUtils, any other issues with this PR?

**Comment by mrdoob** [comment]
@donmccurdy I think I would put it in `GLTFExporter` for now.

**Comment by donmccurdy** [comment]
Ok, I've made the changes. @takahirox this could probably use another round of review when you have the chance. I put the new functions onto `GLTFExporter.Utils.*` so the unit tests can access them; the tests were pretty helpful for this code.

File: examples/js/exporters/GLTFExporter.js [hunk 2]
```diff
@@ -1907,3 +1900,196 @@ THREE.GLTFExporter.prototype = {
 	}
 
 };
+
+THREE.GLTFExporter.Utils = {
+
+	insertKeyframe: function ( track, time ) {
+
+		var tolerance = 0.001; // 1ms
```
Comment (fernandojsg) [inline examples/js/exporters/GLTFExporter.js]: Can you explain the logic behind this number or is it one just smaller enough to be used as epsilon for the comparison? I'm thinking about the timers precisions changes with the meltdown/spectre and not sure if it has something to do with that value.

**Review by fernandojsg — COMMENTED** [review]
Just added a comment out of curiosity, the rest of the code lgtm, and I prefer too to have it directly on the GLTFExporter.Utils until we will have modules available on /examples.
Nice job on the tests too, really useful for this kind of issues :+1:

File: examples/js/exporters/GLTFExporter.js (hunk 2 above)
Comment (donmccurdy) [reply examples/js/exporters/GLTFExporter.js]: Nothing that fancy, just an epsilon small enough to be less than the difference between frames (e.g. at 24-30 fps) and large enough to ignore floating point differences between frame values. :)

Comment (fernandojsg) [reply examples/js/exporters/GLTFExporter.js]: Cool thanks!

**Review by fernandojsg — APPROVED** [review]

**Review by takahirox — APPROVED** [review]
Hm, I'm still in favor of putting Animation clip operation in `AnimationUtil` but yeah we could move another chance if necessary.

BTW I couldn't check functionality because of lack of time...

**Comment by donmccurdy** [comment]
IMO using AnimationUtils would be an easier decision if it weren't in `src/*`. Maybe moving it out or splittings it would make sense eventually. In the meantime lets go with `GLTFExporter.Utils` here.

Thanks @takahirox and @fernandojsg!

**Pull request merged**